src/HexagonOptimize.cpp

```diff
@@ -679,6 +681,27 @@ class OptimizePatterns : public IRMutator {
                 expr = mutate(op->a + neg_b);
                 return;
             } else {
+                // No widening subtracts for the expressions of the form:
```

Comment:
Direct pattern matching is difficult to generate vmpa's [as evident from the code and the comment at L466]. In order to get to a form where we have the args for add_2mpy we'll have to copy the same logic used for vmpa/vdmpy.